Update README.md (#8)
nventive/Chinook.DynamicMvvm: README.md

﻿# Chinook.DynamicMvvm
<!-- ALL-CONTRIBUTORS-BADGE:START - Do not remove or modify this section -->
[![All Contributors](https://img.shields.io/badge/all_contributors-3-orange.svg?style=flat-square)](#contributors-)
<!-- ALL-CONTRIBUTORS-BADGE:END -->

The `Chinook.DynamicMvvm` library assists in MVVM (Model - View - ViewModel) development.

There are 3 main components to `Chinook.DynamicMvvm`:

> * [IViewModel](src/DynamicMvvm.Abstractions/ViewModel/IViewModel.md): a store of `IDisposable` that can be bound to a view and is meant to contain a set of `IDynamicProperties` and `IDynamicCommands`.

Reviewer: Instead of "a store of IDisposable", I think we should focus on something that has properties and commands and implements `INotifyPropertyChanged` because VMs are mainly used for bindings.

* [IDynamicProperty](src/DynamicMvvm.Abstractions/Property/IDynamicProperty.md): a property that will notify its subscribers when its value changes.
* [IDynamicCommand](src/DynamicMvvm.Abstractions/Command/IDynamicCommand.md): a `ICommand` that will notify its subscribers when it is executing.

[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](LICENSE)

## Getting Started

Add the `Chinook.DynamicMvvm` nuget package and its dependencies to your project.

> If your project uses `Reactive` concepts, we recommend installing `Chinook.DynamicMvvm.Reactive` as well, to gain access to `IDynamicProperty` and `IDynamicCommand` extension methods.

Reviewer: Instead of "Reactive concepts", I think we can mention Reactive extensions (as a link) directly.

> If your project uses `Reactive` concepts, we recommend installing `Chinook.DynamicMvvm.Reactive` as well, to gain access to `IDynamicProperty` and `IDynamicCommand` extension methods.

Reviewer 2: ```suggestion
If you want to use [Reactive Extensions](https://github.com/dotnet/reactive), we recommend installing `Chinook.DynamicMvvm.Reactive` to gain access to relevant `IDynamicProperty` and `IDynamicCommand` extension methods
```

> If your project uses `Fluent` validations, we recommend installing `DynamicMvvm.FluentValidation` as well, to gain access to `IViewModel` extension methods.

Reviewer 2: ```suggestion
If you want to use `Fluent` validations, we recommend installing `DynamicMvvm.FluentValidation` to gain access to `IViewModel` extension methods.
```

> If you want to use `Fluent` validations, we recommend installing `DynamicMvvm.FluentValidation` to gain access to `IViewModel` extension methods.

Reviewer: Same here, I'd use the link to FluentValidation https://fluentvalidation.net/

We also recommend installing these [code snippets](https://github.com/nventive/Chinook.DynamicMvvm/tree/master/src/DynamicMvvm.CodeSnippets).

## Features

Please refer to the documentation for [IViewModel](src/DynamicMvvm.Abstractions/ViewModel/IViewModel.md#Features), [IDynamicProperty](src/DynamicMvvm.Abstractions/Property/IDynamicProperty.md#Features) and [IDynamicCommand](src/DynamicMvvm.Abstractions/Command/IDynamicCommand.md#Features) for a full list of the features! Here's a small code sample showing the interaction between `IViewModel`, `IDynamicProperty` and `IDynamicCommand`:

```csharp
public class MyViewModel : ViewModelBase
{
  // This will create and attach a new IDynamicProperty named "MyProperty"
  // to your IViewModel. It's initial value will be default(int).
  public int MyProperty => this.Get<int>();

  // This will create and attach a new IDynamicCommand named "MyCommand"
  // to your IViewModel. It will call the Execute method when executed.
  public IDynamicCommand MyCommand => this.GetCommand(Execute);
  private void Execute() { }
}

> ```

Reviewer 2: I think it would be interesting to show an async command and async property in this sample.
